content/help/faq/traffic-management/viewing-current-rules.md

```diff
@@ -3,4 +3,4 @@ title: How can I view the current route rules I have configured with Istio?
 weight: 10
 ---
 
-Rules can be viewed using `istioctl get routerules -o yaml` or `kubectl get routerules -o yaml`.
+Rules can be viewed using `kubectl get routerules -o yaml`
```

Comment:
Ouch, this is outdated ... should be `kubectl get virtualservices -o yaml`, but this does bring up the question of whether its better to leave off the `-o yaml`? `istioctl get virtualservices` is pretty good these days, so the question is how good is the output from `kubectl get virtualservices`? Do we need to change the kube prereq for Istio to 1.10, to pick up the API resource categories now?

Reply:
We put nice defaults into _istioctl get_.  I agree it would be nice to stick with _kubectl_ but I have not found a way to get good column defaults.  Until that happens I recommend sticking with _istioctl_.

```
VIRTUAL-SERVICE NAME   GATEWAYS           HOSTS     #HTTP     #TCP      NAMESPACE   AGE
bookinfo               bookinfo-gateway   *             1        0      default     35d
~/go/src/istio.io/istio$ kubectl get virtualservice
NAME       AGE
bookinfo   35d
kubectl get virtualservice -o=custom-columns=NAME:.metadata.name,GATEWAYS:spec.gateways,HOSTS:spec.hosts,NAMESPACE:.metadata.namespace
NAME       GATEWAYS             HOSTS     NAMESPACE
bookinfo   [bookinfo-gateway]   [*]       default
```

Reply:
The default `istioctl get` output is nice. I'd argue that `kubectl` still provides a better _overall_ UX with a richer set of features (e.g. `apply`, `edit`, `-o jsonpath`, directory and URL support, `--watch`). We already use `kubectl`extensively through the tasks for most everything else. And there's been confusion in the past about when to use `kubectl` vs. `istioctl`. Sticking with kubectl consistently throughout the docs avoids that confusion.

Longer term I don't think this is a big deal as default server-side columns are available in k8s 1.11 (see https://github.com/istio/istio.github.io/issues/1843#issuecomment-407947035). 

While not ideal, `-o custom-columns-file` is a bit better than `-o custom-columns` in the short term. 

```bash
$ kubectl get virtualservice -o custom-columns-file=$HOME/.kube/template-virtual-service'
NAME      GATEWAYS   HOSTS       NAMESPACE
ratings   <none>     [ratings]   default
reviews   <none>     [reviews]   default

$ cat $HOME/.kube/template-virtual-service
NAME            GATEWAYS        HOSTS           NAMESPACE
.metadata.name  .spec.gateways  .spec.hosts     .metadata.namespace
```